packages/react/src/components/ListBuilder/ListBuilder.jsx

```diff
@@ -22,113 +28,191 @@ export const ListBuilderItemPropTypes = {
+  /** true if the reset button should be shown */
+  hasReset: PropTypes.bool,
   i18n: PropTypes.shape({
     /** (count) => `Items (${count} available)` */
     allListTitle: PropTypes.func,
-
     /** (count) => `${count} items selected` */
     selectedListTitle: PropTypes.func,
-
-    /** remove aria label on selected items */
-    removeLabel: PropTypes.string,
-
+    /** remove icons description on selected items */
+    removeIconDescription: PropTypes.string,
     /** add aria label on unselected items */
     addLabel: PropTypes.string,
-
     /** placeholder text for the search box for all items */
     allListSearchPlaceholderText: PropTypes.string,
-
     /** placeholder text for the search box for selected items */
     selectedListSearchPlaceholderText: PropTypes.string,
-
-    /** expand aria label when using nested groups */
-    expand: PropTypes.string,
-
-    /** close aria label when using nested groups */
-    close: PropTypes.string,
+    /** expand icon description when using nested groups */
+    expandIconDescription: PropTypes.string,
+    /** collapse icon description when using nested groups */
+    collapseIconDescription: PropTypes.string,
+    /** reset button label */
+    resetLabel: PropTypes.string,
+    /** label for the "Load more" button */
+    loadMoreButtonLabel: PropTypes.string,
+    /** message shown if the all items list is empty */
+    allListEmptyText: PropTypes.string,
+    /** message shown if selected list is empty */
+    selectedListEmptyText: PropTypes.string,
+    /** icon description label for the clear search icon */
+    clearSearchIconDescription: PropTypes.string,
   }),
+  /** if true checkboxes will be used for selection so selected items can stay in the left list */
+  removeIcon: ButtonIconPropType,
+  /** the editing style of the hirerchy list showing the selected items */
+  selectedEditingStyle: PropTypes.oneOf([EditingStyle.Single, EditingStyle.SingleNesting]),
 };
 
 const defaultProps = {
+  getAllowedDropIds: null,
+  handleLoadMore: undefined,
+  hasItemsSearch: true,
+  hasSelectedItemsSearch: true,
+  hasReset: false,
   testId: 'list-builder',
-
   items: [],
-
+  itemsSearchValue: null,
   itemCount: null,
-
+  itemWillMove: () => true,
+  loadingMoreIds: [],
+  lockedIds: [],
   selectedItems: [],
-
+  selectedDefaultExpandedIds: [],
   onAdd: null,
-
+  onSelectedListReordered: () => {},
   onRemove: null,
-
+  onReset: () => {},
+  onItemsSearchChange: null,
   i18n: {
     allListTitle: (count) => {
       return `Items (${count} available)`;
     },
     selectedListTitle: (count) => {
       return `${count} Selected`;
     },
-    removeLabel: 'Remove item from list',
+    removeIconDescription: 'Remove item from list',
     addLabel: 'Add item to list',
     allListSearchPlaceholderText: 'Enter a value to search all items',
     selectedListSearchPlaceholderText: 'Enter a value to search selected items',
-    expand: 'Expand',
-    close: 'Close',
```

Comment:
Yes, I thought about that. Since we state that experimental components may have changing API with no major bumps I assumed it was ok to modify the labels, but we can still deprecate. @davidicus what do you think?

Reply:
Deprecated old props